paddle/operators/elementwise_op_function.h

```diff
@@ -340,6 +340,13 @@ void ElementwiseGradCompute(const framework::ExecutionContext& ctx) {
     return;
   }
 
+  if (y_dims.size() == 1 && y_dims[0] == 1) {
+    // y is a scalar
+    auto extended_dims = framework::vectorize(x_dims);
+    extended_dims.push_back(1);
+    x_dims = framework::make_ddim(extended_dims);
```

Comment:
In this case, `axis` should be `x->dims().size()`, otherwise, it is wrong. 
So I think this section of the code should be like this:
```
  int axis = ctx.Attr<int>("axis");
  axis = (axis == -1 ? x_dims.size() - y_dims.size() : axis);

  if (y_dims.size() == 1 && y_dims[0] == 1) {
    // y is a scalar
   axis = x_dims.size();
    auto extended_dims = framework::vectorize(x_dims);
    extended_dims.push_back(1);
    x_dims = framework::make_ddim(extended_dims);
  }

 
```

Reply:
If the `axis` is inconsistent with actual data shape, the code shall give an error instead of modifying `axis` in silence. Because we can never know which one is wrong, the `axis` or the data shape.